bioio_ome_zarr/reader.py

```diff
@@ -140,127 +151,212 @@ def _read_immediate(self) -> xr.DataArray:
+
+        Parameters
+        ----------
+        dims : list of str
+            The dimension names in order (e.g. ["T","C","Z","Y","X"]).
+        shape : tuple of int
+            The lengths of each dimension in `dims`.
+        scene : str
+            Identifier for the current scene, used in default channel IDs.
+        channel_names : list of str or None
+            If provided, use these names for the Channel coordinate; otherwise
+            generate default OME channel IDs.
 
-        # Use dims for coord determination
+        Returns
+        -------
+        coords : dict
+            A mapping from dimension name to coordinate values. Only includes
+            entries for Channel if present in `dims`.
+        """
+        coords = {}
         if dimensions.DimensionNames.Channel in dims:
-            # Generate channel names if no existing channel names
             if channel_names is None:
                 coords[dimensions.DimensionNames.Channel] = [
-                    metadata_utils.generate_ome_channel_id(image_id=scene, channel_id=i)
+                    metadata_utils.generate_ome_channel_id(scene, i)
                     for i in range(shape[dims.index(dimensions.DimensionNames.Channel)])
                 ]
             else:
                 coords[dimensions.DimensionNames.Channel] = channel_names
-
         return coords
 
+    def _get_scale_array(self, dims: Tuple[str, ...]) -> List[float]:
+        """
+        Compute combined scale factors for each dimension by merging the
+        overall and per-dataset coordinate transformations.
+
+        Parameters
+        ----------
+        dims : tuple of str
+            The dimension names in order (e.g. ("T","C","Z","Y","X")).
+
+        Returns
+        -------
+        scale : list of float
+            The elementwise product of the global and dataset-specific scales.
+        """
+        ms = self._multiscales_metadata[self._current_scene_index]
+        overall_scale = ms.get(
+            "coordinateTransformations", [{"scale": [1.0] * len(dims)}]
+        )[0]["scale"]
+        ds_scale = ms["datasets"][self._current_resolution_level][
+            "coordinateTransformations"
+        ][0]["scale"]
+        return [o * d for o, d in zip(overall_scale, ds_scale)]
 
-def get_zarr_reader(fs: AbstractFileSystem, path: str) -> ZarrReader:
-    return ZarrReader(ZarrLocation(zarr.storage.FSStore(url=path, fs=fs)))
+    @property
+    def time_interval(self) -> Optional[types.TimeInterval]:
+        """
+        Returns
+        -------
+        sizes: Time Interval
+            Using available metadata, this float represents the time interval for
+            dimension T.
+
+        """
+        try:
+            if dimensions.DimensionNames.Time in self._get_ome_dims():
+                return self._get_scale_array(self._get_ome_dims())[
+                    self._get_ome_dims().index(dimensions.DimensionNames.Time)
```

Comment:
Doubt it is very expensive, but `self._get_ome_dims()` seems to get calculated quite a few times (3 just in this)

Reply:
I think this also raises a larger question about caching of values. When is this worth doing? Should it only apply to properties that scale computationally with larger images?

Reply:
Do we regularly do performance scans of our code?  If we aren't doing that, I think it is hard to speak to the question of caching vs recompute on the edge cases.

Reply:
I'm fine with letting it be less performant for now